base/gmp.jl

```diff
@@ -69,6 +69,96 @@ function __init__()
     end
 end
 
+
+module MPZ
+# wrapping of libgmp functions
+# "output parameters" are labeled x, y, z, and are returned when appropriate
+# constant input parameter are labeled a, b, c
+
+using Base.GMP: BigInt, Limb
+
+gmpz(op::Symbol) = (Symbol(:__gmpz_, op), :libgmp)
+
+sizeinbase(a::BigInt, b) = Int(ccall((:__gmpz_sizeinbase, :libgmp), Csize_t, (Ptr{BigInt}, Cint), &a, b))
+
+for op in (:add, :sub, :mul, :fdiv_q, :tdiv_q, :fdiv_r, :tdiv_r, :gcd, :lcm, :and, :ior, :xor)
+    @eval begin
+        $op(x::BigInt, a::BigInt, b::BigInt) = (ccall($(gmpz(op)), Void, (Ptr{BigInt}, Ptr{BigInt}, Ptr{BigInt}), &x, &a, &b); x)
```

Comment:
I think Julia wrapper functions should try to aim for following idiomatic Julia naming/argument conventions whenever possible, rather than matching C API naming exactly (if you want to be calling the C API directly, you can always use ccall yourself). We have a bit of an unfortunate mix of these styles at the moment, I have a few of the library binding files on my personal to-do list to clean up in this way and be more consistent about. Disparate C libraries aren't going to have any common conventions, but would be nice if our bindings attempted to.